### PR TITLE
improve testing suite, implement new tests

### PR DESCRIPTION
**Pull request by hems**
Here is a Draft PR I created while testing the library and learning more about how it works, please let me know if I'm missing something, I'm happy to improve / revert some of the changes if necessary.

---

UPDATES:

  - Fix title of test "should revert staking at a pool with amount exceeding token cap", which actually expected the test to `NOT` revert

  - Changes REWARD_AMOUNT to 6048 in order to make numbers round during tests

  - Do not transfer wETH to Alice as Alince does not use wETH

  - Initialize tokens and pool before each test as opposed to once at the start and updates all tests accordingly, therefore:
      - makes every test into a independent test
      - improve clarity in each of the tests
      - allow tests to be re-ordered freely
      - allow easily addding or removing tests

  - Adds Bob account
      - allows tests with more than one participant in the pool
      - implement test where bob takes 25% of the pool

  - Remove duplicated tests on wBTC pool
      - some of the tests for wBTC tested exactly the same piece of code
      on two identical contracts which i ( possibly wrongly ) assumed  
      to be unnecessary

 - add solidity-coverage plugin and `npm run coverage` target

### CODE REVIEW

File: .solcover.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  providerOptions: {
+    default_balance_ether: 100 * 10 ** 18,
+    total_accounts: 10
+  },
+  istanbulReporter: ['html']
+};
```
Comment (JoaquinLiHua) [inline .solcover.js:7]: Kindly add a newline after this!

**Review by JoaquinLiHua — CHANGES_REQUESTED** [review]
Small change requested =)

**Comment by JoaquinLiHua** [comment]
Hello, thanks for submitting the PR!

If you'd like to work on tests for the new governance and treasury contracts, or on developing the smart contracts in general, send a PM on discord to either Boosted Barry or BarryNaut =)

**Comment by JoaquinLiHua** [comment]
@hems when you have fixed the minor errors, happy to merge and close this PR

File: .solcover.js (hunk 1 above)
Comment (hems) [reply .solcover.js]: @JoaquinLiHua added!

**Comment by hems** [comment]
> @hems when you have fixed the minor errors, happy to merge and close this PR

just added the new line! 🚀

**Pull request closed without merging**